samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -77,23 +95,80 @@
    */
   public LocalApplicationRunner(SamzaApplication app, Config config) {
     this.appDesc = ApplicationDescriptorUtil.getAppDescriptor(app, config);
-    this.planner = new LocalJobPlanner(appDesc);
+    this.coordinationUtils = getCoordinationUtils(config);
+    getRunId();
+    this.planner = new LocalJobPlanner(appDesc, coordinationUtils, uid, runId);
   }
 
   /**
    * Constructor only used in unit test to allow injection of {@link LocalJobPlanner}
    */
   @VisibleForTesting
-  LocalApplicationRunner(ApplicationDescriptorImpl<? extends ApplicationDescriptor> appDesc, LocalJobPlanner planner) {
+  LocalApplicationRunner(ApplicationDescriptorImpl<? extends ApplicationDescriptor> appDesc, LocalJobPlanner planner, CoordinationUtils coordinationUtils) {
     this.appDesc = appDesc;
     this.planner = planner;
+    this.coordinationUtils = coordinationUtils;
+  }
+
+  private CoordinationUtils getCoordinationUtils(Config config) {
+    JobCoordinatorConfig jcConfig = new JobCoordinatorConfig(config);
+    String coordinationId = new ApplicationConfig(config).getGlobalAppId() + APPLICATION_RUNNER_PATH_SUFFIX;
+    return jcConfig.getCoordinationUtilsFactory().getCoordinationUtils(coordinationId, uid, config);
+  }
+
+  private void getRunId(){
+    Boolean isAppModeBatch = new ApplicationConfig(appDesc.getConfig()).getAppMode() == ApplicationConfig.ApplicationMode.BATCH;
+    if(coordinationUtils == null ||  !isAppModeBatch) {
+      return;
+    }
+
+    runIdLock = coordinationUtils.getReadWriteLock(RUNID_LOCK_ID);
+    if(runIdLock == null) {
+      LOG.warn("Processor {} failed to create the lock for run.id generation", uid);
```

Comment:
better to find more specific info than UID, uid is not very helpful.

Reply:
Actually, I am using uid is several Log statements. pl let me know if i should remove it out of all of them. 
here there are two cases runIdLock could be null -- ZK failure which will anways throw an exception trying to create ZkDistributedReadWriteLock or in Azure where it is not supported. For Azure, we really dont have any other info to provide.

Would it help to provide an elaborate warning -- something like runid lock creation failed and that old run's state (intermediate streams etc) will be used?